murdock/job.py

```diff
@@ -51,6 +52,9 @@ def __init__(
             if self.pr is not None
             else False
         )  # type: ignore[union-attr]
+        self.priority: int = MurdockJob.calculate_priority(
+            config, pr, ref, self.fasttracked
```

Comment:
all these function parameters are already instance attributes. Thus there's no need for `calculate_priority` to be a staticmethod. A regular instance method, private, should be enough `def _priority(self):` and use `self.config`, `self.pr`, `self.ref` and `self.fastracked` inside.

Reply:
Note that an `@property` method could also be used but it would recompute the priority each time it's accessed. Maybe that better I don't know

Reply:
I wasn't sure, can safely use a class method within `__init__()`?
(Also it needs to be public so it can react to label changes.)

Reply:
> can safely use a class method within __init__()?

yes, if the method is accessing already defined/initialized attributes.

>  it needs to be public so it can react to label changes

then I think a property method is more appropriate.

Reply:
a property method would make e.g., "job.priority" actually call the method?

Reply:
yes so each time `job.priority` is accessed the new value is automatically recomputed

Reply:
In that case I'd rather have a class method.
like, if a label changes, I'd get the job object, get it's current priority, get calculate the priority it would have taking the new labels into account, and if they don't match, re-queue:

```
if job.calculate_priority() != job.priority:
  ... requeue
```

Reply:
assuming the pr field of a job is a reference to updated PR data.
if that's not the case, it's easier to have this staticmethod and supply all needed data at the call site.

Reply:
How do you requeue ? In the current state, a new MurdockJob instance is created and the previous matching ones are marked as cancelled.

I also thought that the display order must also be taken into account. This is something that is in another data structure, a simple list. So a sort on priority must be added there.

https://github.com/murdock-ng/murdock/blob/e62397d29e8ffe1ded2bbb2dd0a7f7a0c49ee853/murdock/murdock.py#L561-569

Reply:
I would add re-queueing here:
https://github.com/murdock-ng/murdock/blob/e62397d29e8ffe1ded2bbb2dd0a7f7a0c49ee853/murdock/murdock.py#L478

(if the job is queued, re-calc its priority and compare to the queued one. if it has changed, don't "return" but fall through to "schedule_job()", which would cancel the old job and create a new, sorted into the queue by priority.)

Reply:
I'm a bit lost. What is the use case exactly ?